nornir/plugins/inventory/netbox.py

```diff
@@ -34,19 +35,30 @@ def __init__(
         nb_token = nb_token or os.environ.get(
             "NB_TOKEN", "0123456789abcdef0123456789abcdef01234567"
         )
-        headers = {"Authorization": "Token {}".format(nb_token)}
 
-        # Create dict of hosts using 'devices' from NetBox
-        r = requests.get(
-            "{}/api/dcim/devices/?limit=0".format(nb_url),
-            headers=headers,
-            params=filter_parameters,
-        )
-        r.raise_for_status()
-        nb_devices = r.json()
+        session = requests.Session()
+        session.headers.update({"Authorization": f"Token {nb_token}"})
+        session.verify = ssl_verify
+
+        # Fetch all devices from Netbox
+        # Since the api uses pagination we have to fetch until no next is provided
+
+        url = f"{nb_url}/api/dcim/devices/?limit=0"
```

Comment:
is `limit=0` needed?

Reply:
It is not strictly required. However if not specified then Netbox will return the number of records configured in PAGINATE_COUNT (50 by default). If limit is set to 0 then the Netbox will return the maximum number of records specified by MAX_PAGE_SIZE configuration.
50 is a bit low and would take quite some requests in a reasonable sized inventory.

Reply:
https://netbox.readthedocs.io/en/latest/api/overview/#pagination

Reply:
ok, thanks for the clarification. Merging :)